packages/extension-chakra-storefront/src/pages/checkout/partials/cc-radio-group.jsx

```diff
@@ -23,54 +24,56 @@ const CCRadioGroup = ({
                     {form.formState.errors.paymentInstrumentId.message}
-                </FormErrorMessage>
+                </Field.ErrorText>
             )}
 
-            <RadioCardGroup value={value} onChange={onPaymentIdChange}>
-                <Stack spacing={4}>
-                    <SimpleGrid columns={[1, 1, 2]} spacing={4}>
-                        {customer.paymentInstruments?.map((payment) => {
+            <RadioCardGroup
+                value={value}
+                onValueChange={(selected) => {
+                    // Chakra v3 radio returns the selected id in an object with a value property
+                    onPaymentIdChange(selected.value)
+                }}
+            >
+                <Stack gap={4}>
+                    <SimpleGrid columns={[1, 1, 2]} gap={4}>
+                        {customer.paymentInstruments?.map((payment, index) => {
                             const CardIcon = getCreditCardIcon(payment.paymentCard?.cardType)
                             return (
                                 <RadioCard
                                     key={payment.paymentInstrumentId}
                                     value={payment.paymentInstrumentId}
+                                    isSelected={payment.paymentInstrumentId === value}
                                 >
                                     <Stack direction="row">
                                         {CardIcon && <CardIcon layerStyle="ccIcon" />}
-                                        <Stack spacing={4}>
-                                            <Stack spacing={1}>
-                                                <Text>{payment.paymentCard?.cardType}</Text>
-                                                <Stack direction="row">
-                                                    <Text>
-                                                        &bull;&bull;&bull;&bull;{' '}
-                                                        {payment.paymentCard?.numberLastDigits}
-                                                    </Text>
-                                                    <Text>
-                                                        {payment.paymentCard?.expirationMonth}/
-                                                        {payment.paymentCard?.expirationYear}
-                                                    </Text>
-                                                </Stack>
-                                                <Text>{payment.paymentCard.holder}</Text>
+                                        <ActionCard
+                                            padding={0}
+                                            border="none"
+                                            onRemove={() => {}}
+                                            data-testid={`sf-checkout-payment-option-${index}`}
+                                            removeBtnLabel={'cc_radio_group.action.remove'}
```

Comment:
This button label is meant for `aria-label`. It needs to be a people-friendly description of what the button is for.